integration_tests/helpers/commonHelpers.ts

```diff
@@ -19,5 +28,47 @@ async function waitForTrue(
   });
 }
 
+async function generateTestEventsAndValidateEventExists(
+  numberOfTestEvents: number,
+  eventName: string,
+  clientId: string
+): Promise<void> {
+  for (let i = 0; i < numberOfTestEvents; i++) {
+    snsValidEventPayload.event_name = eventName;
+    snsValidEventPayload.event_id = generateRandomNumber();
+    details.push(snsValidEventPayload.event_id); // storing event_ids in array to delete from s3 later on
+    snsValidEventPayload.client_id = clientId;
+    await publishSNS(snsValidEventPayload);
+
+    const checkEventId = async (): Promise<boolean> => {
+      const result = await getS3ItemsList(`${prefix}-storage`, objectsPrefix);
+      if (result.Contents !== undefined) {
+        return JSON.stringify(result.Contents.map((data) => data.Key)).includes(
+          snsValidEventPayload.event_id
+        );
+      } else {
+        console.log("Storage bucket contents empty");
+        return false;
+      }
```

Comment:
```suggestion
    if (result.Contents === undefined) {
      console.log("Storage bucket contents empty");
      return false;
    }
    return JSON.stringify(result.Contents.map((data) => data.Key)).includes(
      snsValidEventPayload.event_id
    );
```

This is another one of those denestable statements I mentioned yesterday

Reply:
yes thanks will change it